src/Core/src/VisualDiagnostics/Extensions/VisualDiagnosticsExtensions.Android.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Graphics;
+using Android.Views;
+using GL = Android.Opengl;
+
+namespace Microsoft.Maui
+{
+	public static class VisualDiagnosticsAndroidExtensions
+	{
+		internal static ViewTransform? GetViewTransform(this IView view)
+		{
+			var nativeView = view.GetNative(true);
+			if (nativeView == null)
+				return null;
+			return GetViewTransform(nativeView);
+		}
+
+		internal static ViewTransform? GetViewTransform(View view)
+		{
+			if (view == null || view.Matrix == null || view.Matrix.IsIdentity)
+				return null;
+
+			var m = new float[16];
+			var v = new float[16];
+			var r = new float[16];
+
+			GL.Matrix.SetIdentityM(r, 0);
+			GL.Matrix.SetIdentityM(v, 0);
+			GL.Matrix.SetIdentityM(m, 0);
+
+			GL.Matrix.TranslateM(v, 0, view.Left, view.Top, 0);
+			GL.Matrix.TranslateM(v, 0, view.PivotX, view.PivotY, 0);
+			GL.Matrix.TranslateM(v, 0, view.TranslationX, view.TranslationY, 0);
+			GL.Matrix.ScaleM(v, 0, view.ScaleX, view.ScaleY, 1);
+			GL.Matrix.RotateM(v, 0, view.RotationX, 1, 0, 0);
+			GL.Matrix.RotateM(v, 0, view.RotationY, 0, 1, 0);
+			GL.Matrix.RotateM(m, 0, view.Rotation, 0, 0, 1);
+
+			GL.Matrix.MultiplyMM(r, 0, v, 0, m, 0);
+			GL.Matrix.TranslateM(m, 0, r, 0, -view.PivotX, -view.PivotY, 0);
+
+			return new ViewTransform
+			{
+				M11 = m[0],
+				M12 = m[1],
+				M13 = m[2],
+				M14 = m[3],
+				M21 = m[4],
+				M22 = m[5],
+				M23 = m[6],
+				M24 = m[7],
+				M31 = m[8],
+				M32 = m[9],
+				M33 = m[10],
+				M34 = m[11],
+				OffsetX = m[12],
+				OffsetY = m[13],
+				OffsetZ = m[14],
+				M44 = m[15]
+			};
+		}
+
+		internal static Task<byte[]?> RenderAsPng(this IView view)
```

Comment:
We should move this code to the Essentials Screenshot API and add a overload that takes a native view element. This can just be on the native implementations and we can also add something to maui as an extension method:

```cs
var result = myMauiView.ScreenshotAsync();
```

```cs
ScreenshotResult CaptureScreenshotAsync(this IView view) {
    var nativeView = view.GetNative();
    return Screenshot.CaptureAsync(nativeView);
}
```

We already have duplicated code in the tests, so just moving this code (which appears to have some more checks/features) to essentials and then everyone uses that. And, then we make sure our API is actually useful.

Reply:
I've removed this for now, since there is going to be more work adding this to Essentials, since it needs to reference Core. It's out of scope for now, it's better just to remove this anyway, even if it was internal.